examples/CryptoHamster/contracts/crypto-hamsters.aes

```diff
@@ -11,37 +11,37 @@ contract CryptoHamster =
 
     public stateful function init() = 
         { index = 1,
-          hamsters = {}}
+          map_hamsters = {}}
 
-    public function createHamster(hamsterName: string) =
-        require(!nameExists(hamsterName), "Name is already taken")
+    public function create_hamster(hamster_name: string) =
+        require(!name_exists(hamster_name), "Name is already taken")
 
-        createHamsterByNameDNA(hamsterName, generateRandomDNA())
+        create_hamster_by_name_dna(hamster_name, generate_random_dna())
 
-    public function nameExists(name: string) : bool =
-        switch(Map.lookup(name, state.hamsters))
+    public function name_exists(name: string) : bool =
+        switch(Map.lookup(name, state.map_hamsters))
             None    => false
             Some(x) => true
 
-    public function getHamsterDNA(_name: string) : int =
-        require(nameExists(_name), "There is no hamster with that name!")
+    public function get_hamster_dna(name: string) : int =
+        require(name_exists(name), "There is no hamster with that name!")
 
-        let neededHamster : hamster = state.hamsters[_name]
+        let needed_hamster : hamster = state.map_hamsters[name]
 
-        neededHamster.dna
+        needed_hamster.dna
 
-    private function createHamsterByNameDNA(_name: string, _dna: int) =
-        let newHamster : hamster = {
+    private function create_hamster_by_name_dna(name: string, dna: int) =
```

Comment:
This can be simplified by 

```
private function create_hamster_by_name_dna(name: string, dna: int) =
        put(state{map_hamsters[name] = {id = state.index, name = name, dna = dna}})
        put(state{index = (state.index + 1)})
```

without the additional `new_hamster` variable.

Reply:
for me, this is harder to read

Reply:
Okay, leave it like this then. It's not mandatory to change and probably if the `record` object is more complicated it will be better to do it as you did.